Makefile

```diff
@@ -24,136 +24,14 @@ VNDR=$(shell which vndr || echo '')
 
 GO_LDFLAGS=-ldflags "-X `go list ./version`.Version=$(VERSION)"
 
-.PHONY: clean all AUTHORS fmt vet lint build binaries test integration setup generate protos checkprotos coverage ci check help install uninstall dep-validate
-.DEFAULT: default
+# stop here. do we want to run everything inside of a container, or do we want
+# to run it directly on the host? if the user has set ANY non-empty value for
+# the variable SWARMKIT_USE_CONTAINER, then we do all of the making inside of
+# a container. We will default to using no container, to avoid breaking
+# anyone's workflow
+ifndef SWARMKIT_USE_CONTAINER
+include direct.mk
+else
+include containerized.mk
+endif
```

Comment:
for the record, i cannot believe that conditionally including files in make works.